### DISCUSSION TITLE
Fast Server reload for a ML app containing deep learning model.

### DISCUSSION THREAD

**Discussion by dheerajiiitv** [fastapi/fastapi#8732] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
None
```

### Description

When we are building a ML app containing some big size models we often face issue with server starting time, as it takes a long time to load the model. 

### Wanted Solution

I would like to have a cache system which will keep the model in memory in case of stat reload and will server will reference back to it on restart to save loading time of model.

### Wanted Code

```python
from fastapi import keep_cache

@keep_cache.on_restart
model = classfier.load("location/")
```

### Alternatives

None

### Operating System

Linux

### Operating System Details

Currently using Ubuntu18.04

### FastAPI Version

0.70.0

### Python Version

python3.6

### Additional Context

None

**Comment by ahnaf-zamil** [comment, marked as answer]
As far as I know, the reload mechanism is in the Uvicorn webserver, not FastAPI. FastAPI/Starlette does not come with any integrated web-server, and it is dependent on an ASGI server (example: Uvicorn). So if you want that feature to cache something for restarting, I highly doubt that it's something related to FastAPI; I think it's more related to a specific ASGI web server that you are using.

**Comment by dheerajiiitv** [comment]
Thanks @ahnaf-zamil  I am currently exploring ways to solve this problem in FastAPI or (in uvicorn ASGI server). Let me know if you have any suggestion how we can solve it in FastAPI (or Uvicorn ASGI) or is impossible to implement.
